pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -99,8 +99,15 @@ func (b *Builder) BuildHTTP(class networking.ListenerClass) []*hcm.HttpFilter {
 		// Only applies to inbound and gateways
 		return nil
 	}
+	if b.proxy.SupportsEnvoyExtendedJwt() {
+		filter := b.applier.JwtFilter(true, b.proxy.Type != model.SidecarProxy)
```

Comment:
why do we clear cache only for no sidecar? can we always set it true

And actually envoy only clear when ClearRouteCache = true and `claim_to_headers` added

Reply:
It only makes sense on gateways because there are routes there. Inbound listener has no routes, see https://istio.io/latest/docs/tasks/security/authentication/jwt-route/ disclaimer at the top.

Reply:
IC，i mean once we add another type, this could be broken. Is there any side-effect if we set it true for sidecar

Reply:
Yes, it causes overhead to reset a cache. It is also backwards incompatible since other filters might affect the route as well.